docs/documentation/dependent-resources.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Dependent Resources Feature description: Dependent Resources Feature layout: docs permalink:
+/docs/dependent-resources
+---
+
+# Dependent Resources
+
+DISCLAIMER: The Dependent Resource support is relatively new and, while we strove to cover what we
+anticipate will be the most common use cases, the implementation is not simple and might still
+evolve. As a result, some APIs could be still a subject of change in the future. However,
+non-backwards compatible changes are expected to be trivial to adapt to.
+
+## Motivations and Goals
+
+Most operators need to deal with secondary resources when trying to realize the desired state
+described by the primary resource it is in charge of. For example, the Kubernetes-native
+`Deployment` controller needs to manage `ReplicaSet` instances as part of a `Deployment`'s
+reconciliation process. In this instance, `ReplicatSet` is considered a secondary resource for
+the `Deployment` controller.
+
+Controllers that deal with secondary resources typically need to perform the following steps, for
+each secondary resource:
+
+```mermaid
+flowchart TD
+
+A{Secondary resource exists?}
+A -- Yes --> match
+A -- No --> compute
+
+compute[Compute desired state based on primary state] --> Create --> Done
+match{Matches desired state as defined by primary?}
+match -- Yes --> Done
+match -- No --> Update 
```

Comment:
No it's not actually. In the update case, the full desired state is only computed if the matcher didn't do it as it's often possible to check if the secondary resources match without computing the full desired state.

Reply:
> No it's not actually. In the update case, the full desired state is only computed if the matcher didn't do it as it's often possible to check if the secondary resources match without computing the full desired state.

Yes but to do Create or Update, you need to have a desired state in both case.
If the flow shows when the creatable desired state is computed, it should show when it is computed for Update branch.

Update is slightly more complex due to Matcher computed result can optionally provide a desired state. I thought it would make the whole picture a bit too detailed for an introduction.

Reply:
> No it's not actually. In the update case, the full desired state is only computed if the matcher didn't do it as it's often possible to check if the secondary resources match without computing the full desired state.

Yes but to do Create or Update, you need to have a desired state in both case.
If the flow shows when the creatable desired state is computed, it should show when it is computed for Update branch.

Update is slightly more complex due to Matcher computed result can optionally provide a desired state. I thought it would make the whole picture a bit too detailed for an introduction.

Reply:
That's part of the subtleties of getting the management of dependents right… 😉 
But, indeed, it's probably too detailed for documentation purposes.